### DISCUSSION TITLE
I want to use a column named "type" and set a CHECK constraint on it.

### DISCUSSION THREAD

**Discussion by to-Kenji** [sqldef/sqldef#616] [Feature requests]
Hello. Thanks for the great tool.

I want to use a column named "type" and set a CHECK constraint on it.
However, the first DDL execution succeeds, but subsequent commands(such as exporting or applying changes of table) produce errors.

The error does not occur if I remove the CHECK constraint or change the column name to something else.
Would it be difficult to use a column called "type" and a CHECK constraint on it at the same time?

postgres:14
psqldef: v0.14.5

## Input SQL
```
-- executing this succeeds
CREATE TABLE "sample_table" (
    "type" TEXT NOT NULL DEFAULT 'DEFAULT' CHECK("type" IN ('DEFAULT', 'ANOTHER'))
)
```
## --export output
```
found syntax error when parsing DDL "CREATE TABLE "public"."sample_table" (
    "type" text NOT NULL DEFAULT 'DEFAULT'::text CONSTRAINT sample_table_type_check CHECK (type = ANY (ARRAY['DEFAULT'::text, 'ANOTHER'::text]))
)": syntax error at position 135 near 'type'
```

## Expected output
no error

**Comment by k0kubun** [comment]
> Would it be difficult to use a column called "type" and a CHECK constraint on it at the same time?

There's probably no blocker. It's just that nobody has worked on supporting the syntax.

Would you be interested in filing a pull request to support this? For psqldef, we're switching from `parser/parser.y` to a wrapper of PostgreSQL's original parser so that we can stop writing an actual parser. We just convert a parsed AST parsed by the PostgreSQL parser to sqldef's AST in `database/postgres/parser.go`. When it does not support the conversion, it fallbacks to sqldef's parser in `parser/parser.y`, which is what's currently happening for that syntax.

**Comment by to-Kenji** [comment]
I'm interested, but unfortunately I don't have enough time right now, so I'll try to deal with it without the constraint.
If there is an opportunity in the future, I would like to participate in the support!!
Thank you.

**Comment by gfx** [comment]
The current version of sqldef can recognize `type` as column names. Closing.
